crates/emitter/src/emitter.rs

```diff
@@ -1411,6 +1411,30 @@ impl InstructionWriter {
         self.scope_notes.leave_scope(index, offset);
     }
 
+    pub fn enter_scope_hole(
+        &mut self,
+        maybe_scope_note_index: &Option<ScopeNoteIndex>,
+        parent_scope_note_index: Option<ScopeNoteIndex>,
+    ) -> ScopeNoteIndex {
+        self.debug_leave_lexical_env();
```

Comment:
Can you add `TODO` comment that the bytecode sequence before leaving scope (entering hole) depends on the kind of scope (and also controls)?
this is `debug_leave_lexical_env` because currently there's only simple lexical scope.

https://searchfox.org/mozilla-central/rev/0688ffdef223dac527c2fcdb25560118c4e4df51/js/src/frontend/EmitterScope.cpp#981-1038
https://searchfox.org/mozilla-central/rev/0688ffdef223dac527c2fcdb25560118c4e4df51/js/src/frontend/BytecodeEmitter.cpp#748-807